functionary/prompt_template/prompt_template_v2.py

```diff
@@ -343,6 +343,11 @@ def update_response_state_from_delta_text(
                 "skip_until_reach": self.content_token,  # at first we will skip until reach <|content|>
                 "first_time": True,  # if first_time we return an tempty delta with role=assistant
             }
+        # check if previous token is <|content|>, there might be a space between this token and next token (ex, <|content|> Hello)
+        if current_state["current_text"].endswith(self.content_token):
+            if delta_text[0] == " ":
```

Comment:
We don't exactly know if this is an intended output or a tokenization issue.
For example, a user might ask: "Write me this string ten times: ' I love you '. Then the output will start with empty space and get discarded. Any way to prevent this?

Reply:
Oh good question, I will check this

Reply:
> We don't exactly know if this is an intended output or a tokenization issue. For example, a user might ask: "Write me this string ten times: ' I love you '. Then the output will start with empty space and get discarded. Any way to prevent this?

I don't know why i still get prs and all for this in my email,

anyway, This is a artifact of just how BPE tokenizers work. after a special token, it adds a space. long story short, you need that space, otherwise the model will probably throw a fit since it won't tokenize properly.